modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/TrafficSources.js

```diff
@@ -41,27 +43,64 @@ const FALLBACK_COLOR = '#e92563';
 const getColorByName = (name) => COLORS_MAP[name] || FALLBACK_COLOR;
 
 export default function TrafficSources({
+	dataProvider,
 	languageTag,
 	onTrafficSourceClick,
-	trafficSources,
 }) {
 	const [highlighted, setHighlighted] = useState(null);
 
-	const [, addWarning] = useWarning();
+	const [warning, addWarning] = useWarning();
+
+	const {validAnalyticsConnection} = useContext(ConnectionContext);
 
 	const [{publishedToday}] = useContext(StoreContext);
 
-	const fullPieChart = trafficSources.some((source) => !!source.value);
+	const [trafficSources, setTrafficSources] = useState([]);
+
+	const isMounted = useIsMounted();
+
+	useEffect(() => {
+		if (validAnalyticsConnection) {
+			dataProvider()
+				.then((trafficSources) => {
+					if (isMounted()) {
+						setTrafficSources(trafficSources);
+					}
+				})
+				.catch(() => {
+					setTrafficSources([]);
+					if (isMounted()) {
+						if (!warning) {
+							addWarning();
+						}
+					}
+				});
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
What did it want you to add/remove here? It might be something that you could harmlessly add.

Reply:
You mean remove that commented line and add it into the dependencies? It's missing `isMounted` and `addWarning`. Should I added it? 🤔

Reply:
Yeah, that's what I meant.

- `isMounted` should be stable [because it comes from `useCallback`](https://github.com/liferay/liferay-portal/blob/31885f9c46205904787c5d23cd5f804f0f84916f/modules/apps/frontend-js/frontend-js-react-web/src/main/resources/META-INF/resources/js/hooks/useIsMounted.es.js#L37).
- `addWarning` also, [for the same reason](https://github.com/liferay-frontend/liferay-portal/blob/875af1be3012d41367503b4e493a1c8c6b915247/modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/context/store.js#L68-L72).

That means you can add them to the dependencies array and they won't cause unnecessary re-evaluations, because they won't change. Also, comparing them is super cheap (React uses `Object.is` on items in the dependencies array), which means that having them in the dependencies array is basically "free", and it allows you to get rid of an ugly lint suppression.